src/ast/nodes/BinaryExpression.ts

```diff
@@ -37,12 +38,15 @@ export default class BinaryExpression extends NodeBase {
 	right: ExpressionNode;
 	operator: keyof typeof binaryOperators;
 
-	getLiteralValueAtPath(path: ObjectPath, options: ExecutionPathOptions): LiteralValueOrUnknown {
+	getLiteralValueAtPath(
+		path: ObjectPath,
+		getValueTracker: ImmutableEntityPathTracker
```

Comment:
`getValue` makes it sound like this will be a function. How about just calling this `tracker` or even `recursionTracker`? As someone not familiar with the approach, I only finally got the "recursion" part properly when you explained it in the last comment, as otherwise there was nothing in the code to indicate this is what it's for.

Reply:
Yes, `recursionTracker` is probably much clearer to outsiders. Thanks for looking into this that deeply, this really helps to make the code a lot easier to understand!